app/Concerns/ImpactsCategoryCounts.php

```diff
@@ -29,19 +31,60 @@ protected function incrementThreadCount(array $data): bool|array
 
     /**
      * Updates the category's discussion count.
+     *
+     * @throws Exception
      */
     protected function decrementThreadCount(array $data)
     {
         if (! $data['result']) {
             return false;
         }
 
-        $thread = $this->allowCallbacks(false)->onlyDeleted()->find($data['id']);
-
-        // Decrement Category thread count
-        model(CategoryModel::class)->decrementStats($thread[0]->category_id, 'thread_count');
-        // Decrement User thread count
-        model(UserModel::class)->decrementStats($thread[0]->author_id, 'thread_count');
+        // The $data['id'] is always an array on delete
+        $threads = $this->allowCallbacks(false)->onlyDeleted()->find($data['id']);
+
+        foreach ($threads as $thread) {
+            $postModel = model(PostModel::class);
+            // Soft-delete all the posts for this thread
+            $postModel->builder()
+                ->update(
+                    ['deleted_at' => $thread->deleted_at],
+                    ['thread_id' => $thread->id, 'deleted_at' => null]
```

Comment:
Shouldn't `deleted_at`  be added to `$allowedFields`?
https://github.com/lonnieezell/forum-example/blob/b4010a8c72a33e64cb9a93c9f656cb26fa387beb/app/Models/PostModel.php#L25-L27

Reply:
For now, I don't think so. There is no scenario where we will be updating this field with a model.
All stats-related stuff is made with the use of a builder instance.

Reply:
Okay,